src/axom/quest/interface/inout.cpp

```diff
@@ -283,103 +288,164 @@ struct InOutHelper
   GeometricBoundingBox m_meshBoundingBox;
   SpacePt m_meshCenterOfMass;
 
-  Parameters m_params;
+  InOutParameters m_params;
   State m_state;
 };
 }  // end namespace internal
 
 /// Static instance of the InOutHelper in 3D.
 /// Used by the quest::inout API functions
-static internal::InOutHelper<3> s_inoutHelper;
+static internal::InOutHelper<3> s_inoutHelper3D;
+
+/// Static instance of the InOutHelper in 2D
+/// Used by the quest::inout API functions
+static internal::InOutHelper<2> s_inoutHelper2D;
 
-bool inout_initialized() { return s_inoutHelper.isInitialized(); }
+static internal::InOutParameters s_inoutParams;
+
+//------------------------------------------------------------------------------
+
+bool inout_initialized()
+{
+  const int dim = inout_get_dimension();
+  return (dim == 2) ? s_inoutHelper2D.isInitialized()
+                    : s_inoutHelper3D.isInitialized();
+}
 
 int inout_init(const std::string& file, MPI_Comm comm)
 {
+  const int dim = inout_get_dimension();
+  int rc = QUEST_INOUT_FAILED;
+
   if(inout_initialized())
   {
     SLIC_WARNING("quest inout query already initialized ");
-    return QUEST_INOUT_FAILED;
+    return rc;
   }
 
-  int rc = s_inoutHelper.initialize(file, comm);
-  if(rc == QUEST_INOUT_FAILED)
+  switch(dim)
   {
-    s_inoutHelper.restoreLoggingLevel();
+  case 2:
+    s_inoutHelper2D.setVerbose(s_inoutParams.m_verbose);
+    s_inoutHelper2D.setSegmentsPerKnotSpan(s_inoutParams.m_segmentsPerKnotSpan);
+    s_inoutHelper2D.setVertexWeldThreshold(s_inoutParams.m_vertexWeldThreshold);
+
+    rc = s_inoutHelper2D.initialize(file, comm);
+    if(rc == QUEST_INOUT_FAILED)
+    {
+      s_inoutHelper2D.restoreLoggingLevel();
+    }
+    break;
+
+  case 3:
+    s_inoutHelper3D.setVerbose(s_inoutParams.m_verbose);
+    s_inoutHelper3D.setVertexWeldThreshold(s_inoutParams.m_vertexWeldThreshold);
+
+    rc = s_inoutHelper3D.initialize(file, comm);
+    if(rc == QUEST_INOUT_FAILED)
+    {
+      s_inoutHelper3D.restoreLoggingLevel();
+    }
+    break;
+
+  default:
+    rc = QUEST_INOUT_FAILED;
+    break;
   }
+
   return rc;
 }
 
 int inout_init(mint::Mesh*& mesh, MPI_Comm comm)
 {
+  const int dim = inout_get_dimension();
+  int rc = QUEST_INOUT_FAILED;
+
   if(inout_initialized())
   {
     SLIC_WARNING("quest inout query already initialized ");
     return QUEST_INOUT_FAILED;
   }
 
-  int rc = s_inoutHelper.initialize(mesh, comm);
-  if(rc == QUEST_INOUT_FAILED)
+  switch(dim)
   {
-    s_inoutHelper.restoreLoggingLevel();
-  }
-  return rc;
-}
+  case 2:
+    s_inoutHelper2D.setVerbose(s_inoutParams.m_verbose);
+    s_inoutHelper2D.setSegmentsPerKnotSpan(s_inoutParams.m_segmentsPerKnotSpan);
+    s_inoutHelper2D.setVertexWeldThreshold(s_inoutParams.m_vertexWeldThreshold);
 
-int inout_finalize() { return s_inoutHelper.finalize(); }
+    rc = s_inoutHelper2D.initialize(mesh, comm);
+    if(rc == QUEST_INOUT_FAILED)
+    {
+      s_inoutHelper2D.restoreLoggingLevel();
+    }
```

Comment:
Missing a break here?

Reply:
Indeed!

Double whammy -- I had tests for this function in 3D in the `quest_inout_interface` file, but had not extended them to 2D in this PR. 

I updated the tests to 2D, verified that the missing `break` was caught by the test and then fixed this.

Thanks!